lib/galaxy/managers/library_datasets.py

```diff
@@ -115,7 +121,7 @@ def _validate_and_parse_update_payload(self, payload):
                 validated_payload[key] = val
             if key in ('file_ext'):
                 datatype = self.app.datatypes_registry.get_datatype_by_extension(val)
-                if datatype is None:
+                if datatype is None and val not in ("auto"):
```

Comment:
```suggestion
                if datatype is None and val not in ("auto",):
```

Reply:
How did this pass lint? Thanks.

Reply:
Because it's equivalent to `val not in "auto"` :
```
In [5]: "auto" in "auto"
Out[5]: True
```

Reply:
Python tryin got be helpful, nice.